Sources/SwiftParser/Expressions.swift

```diff
@@ -1911,9 +1915,17 @@ extension Parser {
           arena: self.arena
         )
       )
-    } while keepGoing != nil && self.hasProgressed(&loopProgress)
+    } while keepGoing != nil && !atArgumentListTerminator(allowTrailingComma) && self.hasProgressed(&loopProgress)
     return result
   }
+
+  mutating func atArgumentListTerminator(_ allowTrailingComma: Bool) -> Bool {
+    if allowTrailingComma, self.at(.rightParen) {
+      return true
+    } else {
+      return false
+    }
```

Comment:
I was wondering if wouldn't be better if, instead of writing if/else to return a boolean, if wouldn't be better to write just one line of code, like this:
```suggestion
    allowTrailingComma && self.at(.rightParen)
```